drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -231,32 +321,142 @@ static int stm32_clock_control_init(struct device *dev)
 	while (LL_RCC_HSE_IsReady() != 1) {
 	}
 
-	/* Set FLASH latency */
-	LL_FLASH_SetLatency(LL_FLASH_LATENCY_4);
-
 	/* Main PLL configuration and activation */
 	LL_RCC_PLL_SetSource(LL_RCC_PLLSOURCE_HSE);
+
+	pllsrc_clock = CONFIG_CLOCK_STM32_HSE_CLOCK;
```

Comment:
I suggest it stays as **CONFIG_CLOCK_STM32_HSE_CLOCK** because how would you either provide your custom board external crystal value other than using the Kconfig files ?

HSE_VALUE is defined in **stm32h7xx_hal_conf.h** which, correct me if I'm wrong, shouldn't be user modifiable ?

Reply:
This symbol is defined as follows in cube packages:
```
#if !defined  (HSE_VALUE)
#define HSE_VALUE    8000000U  /*!< Value of the HSE oscillator in Hz */
#endif /* HSE_VALUE */
```
And HSE_VALUE is feed in packages thanks to the following in `modules/hal/stm32/stm32cube/CMakeLists.txt`:
```
if(CONFIG_CLOCK_STM32_HSE_CLOCK)
  zephyr_compile_definitions( -DHSE_VALUE=${CONFIG_CLOCK_STM32_HSE_CLOCK} )
endif()
```
So it is actually configurable.

Reply:
Alright then, there is no reason to put the config symbol here. I'm switching to HSE_VALUE !